src/Maestro/Maestro.Web/Api/v2019_01_16/Controllers/GoalController.cs

```diff
@@ -0,0 +1,97 @@
+using Maestro.Web.Api.v2019_01_16.Models;
+using Microsoft.AspNetCore.ApiVersioning;
+using Microsoft.AspNetCore.ApiVersioning.Swashbuckle;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.KeyVault;
+using Microsoft.EntityFrameworkCore;
+using System.Net;
+
+namespace Maestro.Web.Api.v2019_01_16.Controllers
+{
+    [Route("goals")]
+    [ApiVersion("2019-01-16")]
+    public class GoalController : ControllerBase
+    {
+        protected readonly BuildAssetRegistryContext _context;
+        public GoalController(BuildAssetRegistryContext context)
+        {
+            _context = context;
+        }
+
+        /// <summary>
+        ///   Exposes methods to set goal time <see cref="Goal"/>
+        /// </summary>
+        /// <param name="ChannelName"></param>
+        /// <param name="DefinitionId"></param>
+        /// <param name="Minutes"></param>
+        [HttpPost]
+        [SwaggerApiResponse(System.Net.HttpStatusCode.Created, Type = typeof(Models.Goal), Description = "Goal for given Channel and Definition-Id is created/updated")]
+        [ValidateModelState]
+        public virtual async Task<IActionResult> Create([Required] int DefinitionId, [Required] string ChannelName, [Required] int Minutes)
+        {
+            Data.Models.Channel channel = await _context.Channels
+                    .Where(c => c.Name.Equals(ChannelName)).FirstOrDefaultAsync();
+            Data.Models.GoalTime goal = await _context.GoalTime
+                .Where(g => g.DefinitionId == DefinitionId && g.ChannelId == channel.Id).FirstOrDefaultAsync();
+
+            if (channel == null)
+            {
+                return NotFound();
+            }
+            if (goal == null)
+            {
+                var goalModel = new Data.Models.GoalTime
+                {
+                    DefinitionId = DefinitionId,
+                    Minutes = Minutes,
+                    ChannelId = channel.Id
+                };
+                await _context.GoalTime.AddAsync(goalModel);
+                await _context.SaveChangesAsync();
+                var goalTest = CreatedAtRoute(
+                    new
+                    {
+                        action = "CreateGoal",
+                        DefinitionId = DefinitionId,
+                        Minutes = Minutes,
+                        ChannelId = channel.Id
+                    },
+                    new Goal(goalModel));
+                return StatusCode((int)HttpStatusCode.Created);
+            }
+            else
+            {
+                goal.Minutes = Minutes;
+                _context.GoalTime.Update(goal);
+                await _context.SaveChangesAsync();
+
+                return Ok(new Goal(goal));
+            }
+        }
+
+        /// <summary>
+        /// Get goal time in minutes <see cref="Goal"/>s that match the given classification.
+        /// </summary>
+        /// <param name="ChannelName"></param>
```

Comment:
These should probably have comments in them.

Reply:
I changed the comment here.

Reply:
The parameters. Having empty XML tags is just wasting space.  Either fill in the comment tags with something, or delete them. (I'd prefer filling them in with something useful)

Reply:
added comments to fill in the empty xml tags.